### PR TITLE
Use native ParquetWriter in Delta CheckpointWriter

### PR DESCRIPTION
**Pull request by electrum**
See #16020

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java [hunk 1]
```diff
@@ -122,22 +119,20 @@ public void write(ConnectorSession session, CheckpointEntries entries, Path targ
...
-                jobConf,
-                typeManager,
-                DateTimeZone.UTC,
-                session);
+        ParquetSchemaConverter schemaConverter = new ParquetSchemaConverter(columnTypes, columnNames, false, false);
+
+        ParquetWriter writer = new ParquetWriter(
+                fileSystemFactory.create(session)
+                        .newOutputFile(targetPath)
+                        .create(newSimpleAggregatedMemoryContext()),
```
Comment (dain) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java:127]: Didn't you make memory context optional in another PR?

**Review by dain — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java (hunk 1 above)
Comment (electrum) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java:127]: You're right, this is in the ACID PR. I'll merge that first.
